guides/plugins/plugins/framework/data-handling/entities-via-attributes.md

```diff
@@ -0,0 +1,411 @@
+---
+nav:
+  title: Entities via attributes
+  position: 1
+
+---
+
+# Entities via attributes
+
+Since shopware v6.6.3.0 it is possible to register entities via php attributes. This is a new way to register entities in Shopware 6. This guide will show you how to do this.
+
+## Define the entity
+
+First, you need to define your entity. This is done by creating a new class and adding the `Entity` attribute to it. The attribute requires the `name` parameter, which is the name of the entity. The `name` parameter is required and must be unique.
+
+Also, you have to extend the `EntityStruct` class and define a primary key. The primary key is defined by adding the `PrimaryKey` attribute to a property. In theory, the primary key can be of any type, but it is recommended to use a `UUID`. 
+
+```php
+<?php
+
+namespace Shopware\Tests\Integration\Core\Framework\DataAbstractionLayer\fixture;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Entity as EntityStruct;
+use Shopware\Core\Framework\DataAbstractionLayer\Attribute\Entity;
+
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID)]
+    public string $id;
+}
+```
+
+This is the most basic entity definition. You can add more properties and attributes to the entity. For example, you can add the `Field` attribute to a property to define the type of the property.
+
+## Register the entity
+
+To register the entity, you have to add this class to the DI container in the `services.xml` file. This is done by adding the `shopware.entity` tag to the service definition. 
+
+```xml
+<service id="Shopware\Tests\Integration\Core\Framework\DataAbstractionLayer\fixture\AttributeEntity">
+    <tag name="shopware.entity"/>
+</service>
+```
+
+That's it. You entity is registered and you can read and write data to it over the DAL. Over the tag, shopware automatically registers an `EntityDefinition` for the entity and an `EntityRepository`.
+
+## Field Types
+
+To define more fields, you typically use the `Field` attribute. The `Field` attribute requires the `type` parameter, which is the type of the field. The type can be any of the `FieldType` constants. 
```

Comment:
"typically" ? is there another way? :grin: 

To define more fields, add a new public property and add the `Field` attribute to it.

Reply:
yes here are other ways, other field classes.